configparser.go

```diff
@@ -183,3 +194,119 @@ func (c *ConfigParser) Getbool(section, option string) (val bool, err error) {
 
 	return val, err
 }
+
+// Add the section or return an error if it already exists
+func (c *ConfigParser) AddSection(section string) error {
+	if c.HasSection(section) {
+		return newDuplicateSectionError(section)
+	}
+
+	c.sections[section] = Section{options: map[string]string{}}
+
+	return nil
+}
+
+// Return whether the named section is present in the configuration
+func (c *ConfigParser) HasSection(section string) bool {
+	_, ok := c.sections[section]
+	return ok
+}
+
+// Return whether the named option is present in the configuration
+func (c *ConfigParser) HasOption(section, option string) bool {
+	sect, ok := c.sections[section]
+	if !ok {
+		return false
+	}
+	_, ok = sect.options[option]
+
+	return ok
+}
+
+// Set the given option to the specified value or return an error if the
+// section doesn't exist
+func (c *ConfigParser) Set(section, option, value string) error {
+	if !c.HasSection(section) {
+		if c.AllowNoSectionHeader && section == "" {
+			c.sections[""] = Section{options: map[string]string{}}
+		} else {
+			return newNoSectionError(section)
+		}
+	}
+
+	c.sections[section].options[option] = value
+
+	return nil
+}
+
+// A convenience function to write the current configuration to a file,
+// optionally using spaces around delimiters. This call is not present in
+// the Python ConfigParser.
+func (c *ConfigParser) WriteFile(path string, spaceAroundDelimiters bool, mode os.FileMode) error {
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_TRUNC|os.O_CREATE, mode)
```

Comment:
This can be made more robust by writing to a different filename and renaming once the write operation is complete to avoid overwriting the original file should the operation fail. This is a simple initial implementation because this call is not part of the original API and I don't know if it will be accepted or not.